# Final Project Submission

Please fill out:
* Student name: Aashish Gatamaneni
* Student pace: self paced
* Scheduled project review date/time: 
* Instructor name: Morgan Jones
* Blog post URL:


# Aviation Airplane Safety Analysis


### Place image here

## Business Problem
This project takes a look at the Aviation Accident Dataset which contains data about civil aviation accidents and incidents in the United States and its territories, and international waters. Using this dataset, I identify patterns in accidents that result in the most damage to the aircraft and its passengers to help the Aviation company determine aircrafts that present lower risks for the company.

### Place image here

## Data Understanding

The Aviation Accident Database contains aviation accident data from 1962-2023 with each accident is given a unique ID that displays all the accident and aircraft data associated with that unique ID (e.g. weather, aircraft model, etc.)

In [2]:
import pandas as pd
import numpy as np

In [4]:
aviation = pd.read_csv('./data/Aviation_Data.csv')

/var/folders/tr/25_dhxdn761_fdmhsy7p1r9r0000gn/T/ipykernel_64065/4063860901.py:1: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  aviation = pd.read_csv('./data/Aviation_Data.csv')


In [5]:
aviation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [19]:
#Check for missing values
aviation.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40216
Airport.Name              37644
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2841
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8555
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7843
Publication.Date          16689
dtype: i

In [6]:
aviation.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [7]:
aviation['Total.Fatal.Injuries'].value_counts()

Total.Fatal.Injuries
0.0      59675
1.0       8883
2.0       5173
3.0       1589
4.0       1103
         ...  
156.0        1
68.0         1
31.0         1
115.0        1
176.0        1
Name: count, Length: 125, dtype: int64

In [8]:
aviation['Weather.Condition'].value_counts()

Weather.Condition
VMC    77303
IMC     5976
UNK      856
Unk      262
Name: count, dtype: int64

In [9]:
aviation['Engine.Type'].value_counts()

Engine.Type
Reciprocating      69530
Turbo Shaft         3609
Turbo Prop          3391
Turbo Fan           2481
Unknown             2051
Turbo Jet            703
Geared Turbofan       12
Electric              10
LR                     2
NONE                   2
Hybrid Rocket          1
UNK                    1
Name: count, dtype: int64

In [10]:
aviation['Broad.phase.of.flight'].value_counts()

Broad.phase.of.flight
Landing        15428
Takeoff        12493
Cruise         10269
Maneuvering     8144
Approach        6546
Climb           2034
Taxi            1958
Descent         1887
Go-around       1353
Standing         945
Unknown          548
Other            119
Name: count, dtype: int64

In [11]:
aviation['Report.Status'].value_counts()

Report.Status
Probable Cause                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        61754
Foreign                                                                                                                                                                                                                                                                                       

In [12]:
aviation['Make'].value_counts()

Make
Cessna             22227
Piper              12029
CESSNA              4922
Beech               4330
PIPER               2841
                   ...  
Leonard Walters        1
Maule Air Inc.         1
Motley Vans            1
Perlick                1
ROYSE RALPH L          1
Name: count, Length: 8237, dtype: int64

In [15]:
aviation['Air.carrier'].value_counts()

Air.carrier
Pilot                        258
American Airlines             90
United Airlines               89
Delta Air Lines               53
SOUTHWEST AIRLINES CO         42
                            ... 
WOODY CONTRACTING INC          1
Rod Aviation LLC               1
Paul D Franzon                 1
TRAINING SERVICES INC DBA      1
MC CESSNA 210N LLC             1
Name: count, Length: 13590, dtype: int64

## Data Preparation

### Data Cleaning

#### Dropping columns

First, I'm going the make the columns more easily accessible. We also have a lot of columns that won't be relevant, so I will drop those unnecessary columns

In [88]:
aviation.columns = aviation.columns = aviation.columns.str.lower().str.replace('.', '_')
aviation.drop(columns = ['latitude', 'longitude', 'airport_code', 'airport_name', 'purpose_of_flight', 'publication_date'], inplace = True)
aviation.head()

event_id investigation_type accident_number  event_date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          location        country injury_severity aircraft_damage  \
0  MOOSE CREEK, ID  United States        Fatal(2)       Destroyed   
1   BRIDGEPORT, CA  United States        Fatal(4)       Destroyed   
2    Saltville, VA  United States        Fatal(3)       Destroyed   
3       EUREKA, CA  United States        Fatal(2)       Destroyed   
4       Canton, OH  United States        Fatal(1)       Destroyed   

  registration_number      make  ... number_of_engines    engine_type  \
0              NC6404   Stinson  ...               1.0  Reciprocating   
1              N5069P     Piper  ...               1.0  Reciprocating   
2              N5142R    Cessna  ...               1.0  Reciprocating   
3              N1168J  Rockwell  ...               1.0  Reciprocating   
4               N15NY    Cessna  ...               NaN        Missing   

   total_fatal_injuries total_serious_injuries  total_minor_injuries  \
0                   2.0                    0.0                   0.0   
1                   4.0                    0.0                   0.0   
2                   3.0                    NaN                   NaN   
3                   2.0                    0.0                   0.0   
4                   1.0                    2.0                   NaN   

   total_uninjured  weather_condition  broad_phase_of_flight   report_status  \
0              0.0            Missing                 Cruise  Probable Cause   
1              0.0            Missing                Missing  Probable Cause   
2              NaN                IMC                 Cruise  Probable Cause   
3              0.0                IMC                 Cruise  Probable Cause   
4              0.0                VMC               Approach  Probable Cause   

  publication_date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980  

[5 rows x 22 columns]

I dropped the columns that weren't going to help with observations, but I'm also going to drop the columns that have mostly missing data

In [30]:
aviation.isna().sum()

event_id                   1459
investigation_type            0
accident_number            1459
event_date                 1459
location                   1511
country                    1685
injury_severity            2459
aircraft_damage            4653
aircraft_category         58061
registration_number        2841
make                       1522
model                      1551
amateur_built              1561
number_of_engines          7543
engine_type                8555
far_description           58325
schedule                  77766
air_carrier               73700
total_fatal_injuries      12860
total_serious_injuries    13969
total_minor_injuries      13392
total_uninjured            7371
weather_condition          5951
broad_phase_of_flight     28624
report_status              7843
publication_date          16689
dtype: int64

We can drop the **aircraft_category**, **far_description**, **schedule**, and **air_carrier** columns which all have more missing data than actual data

In [31]:
aviation.drop(columns = ['aircraft_category', 'far_description', 'schedule', 'air_carrier'], inplace = True)
aviation.head()

event_id investigation_type accident_number  event_date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          location        country injury_severity aircraft_damage  \
0  MOOSE CREEK, ID  United States        Fatal(2)       Destroyed   
1   BRIDGEPORT, CA  United States        Fatal(4)       Destroyed   
2    Saltville, VA  United States        Fatal(3)       Destroyed   
3       EUREKA, CA  United States        Fatal(2)       Destroyed   
4       Canton, OH  United States        Fatal(1)       Destroyed   

  registration_number      make  ... number_of_engines    engine_type  \
0              NC6404   Stinson  ...               1.0  Reciprocating   
1              N5069P     Piper  ...               1.0  Reciprocating   
2              N5142R    Cessna  ...               1.0  Reciprocating   
3              N1168J  Rockwell  ...               1.0  Reciprocating   
4               N15NY    Cessna  ...               NaN            NaN   

   total_fatal_injuries total_serious_injuries  total_minor_injuries  \
0                   2.0                    0.0                   0.0   
1                   4.0                    0.0                   0.0   
2                   3.0                    NaN                   NaN   
3                   2.0                    0.0                   0.0   
4                   1.0                    2.0                   NaN   

   total_uninjured  weather_condition  broad_phase_of_flight   report_status  \
0              0.0                UNK                 Cruise  Probable Cause   
1              0.0                UNK                Unknown  Probable Cause   
2              NaN                IMC                 Cruise  Probable Cause   
3              0.0                IMC                 Cruise  Probable Cause   
4              0.0                VMC               Approach  Probable Cause   

  publication_date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980  

[5 rows x 22 columns]

#### Check for duplicates

In [32]:
duplicates = aviation[aviation.duplicated()]
print(len(duplicates))
duplicates.head()

1390


event_id investigation_type accident_number event_date location country  \
64050      NaN         25-09-2020             NaN        NaN      NaN     NaN   
64052      NaN         25-09-2020             NaN        NaN      NaN     NaN   
64388      NaN         25-09-2020             NaN        NaN      NaN     NaN   
64541      NaN         25-09-2020             NaN        NaN      NaN     NaN   
64552      NaN         25-09-2020             NaN        NaN      NaN     NaN   

      injury_severity aircraft_damage registration_number make  ...  \
64050             NaN             NaN                 NaN  NaN  ...   
64052             NaN             NaN                 NaN  NaN  ...   
64388             NaN             NaN                 NaN  NaN  ...   
64541             NaN             NaN                 NaN  NaN  ...   
64552             NaN             NaN                 NaN  NaN  ...   

      number_of_engines engine_type  total_fatal_injuries  \
64050               NaN         NaN                   NaN   
64052               NaN         NaN                   NaN   
64388               NaN         NaN                   NaN   
64541               NaN         NaN                   NaN   
64552               NaN         NaN                   NaN   

      total_serious_injuries  total_minor_injuries  total_uninjured  \
64050                    NaN                   NaN              NaN   
64052                    NaN                   NaN              NaN   
64388                    NaN                   NaN              NaN   
64541                    NaN                   NaN              NaN   
64552                    NaN                   NaN              NaN   

       weather_condition  broad_phase_of_flight report_status publication_date  
64050                NaN                    NaN           NaN              NaN  
64052                NaN                    NaN           NaN              NaN  
64388                NaN                    NaN           NaN              NaN  
64541                NaN                    NaN           NaN              NaN  
64552                NaN                    NaN           NaN              NaN  

[5 rows x 22 columns]

There seems to be 1390 duplicate rows, so I will drop the duplicate rows except for the first occurrence

In [33]:
aviation.drop_duplicates(inplace=True)

In [35]:
duplicates = aviation[aviation.duplicated()]
print(len(duplicates))
duplicates.head()

0


Empty DataFrame
Columns: [event_id, investigation_type, accident_number, event_date, location, country, injury_severity, aircraft_damage, registration_number, make, model, amateur_built, number_of_engines, engine_type, total_fatal_injuries, total_serious_injuries, total_minor_injuries, total_uninjured, weather_condition, broad_phase_of_flight, report_status, publication_date]
Index: []

[0 rows x 22 columns]

I will drop all the observations (rows) that only have missing values

In [51]:
aviation.dropna(subset=['event_id'], inplace = True)
aviation.isna().sum()

event_id                      0
investigation_type            0
accident_number               0
event_date                    0
location                     52
country                     226
injury_severity            1000
aircraft_damage            3194
registration_number        1382
make                         63
model                        92
amateur_built               102
number_of_engines          6084
engine_type                7096
total_fatal_injuries      11401
total_serious_injuries    12510
total_minor_injuries      11933
total_uninjured            5912
weather_condition          4492
broad_phase_of_flight     27165
report_status              6384
publication_date          15230
dtype: int64

In [58]:
aviation[aviation['make'].isna()]

event_id investigation_type accident_number  event_date  \
1767   20020917X02887           Accident      ATL82FLQ07  1982-07-01   
1971   20020917X02861           Accident      ATL82FIG10  1982-07-16   
2458   20020917X03618           Accident      LAX82DA316  1982-08-21   
2507   20020917X03865           Accident      MKC82DA156  1982-08-24   
2733   20020917X04964           Accident      MKC82FA171  1982-09-11   
...               ...                ...             ...         ...   
73385  20130307X10950           Incident      DCA13WA062  2013-03-06   
73735  20130531X43222           Incident      DCA13WA097  2013-05-22   
73761  20130808X52306           Incident      CEN13WA470  2013-05-26   
75113  20140327X61657           Accident      ERA14LA174  2014-03-26   
78394  20160314X33602           Incident      ENG16WA009  2016-03-10   

                  location        country injury_severity aircraft_damage  \
1767         MCWHORTER, KY  United States       Non-Fatal       Destroyed   
1971         FREDERICK, MD  United States        Fatal(1)       Destroyed   
2458           VENTURA, CA  United States       Non-Fatal       Destroyed   
2507            SIDNEY, NE  United States       Non-Fatal     Substantial   
2733        PLATTSBURG, MO  United States       Non-Fatal     Substantial   
...                    ...            ...             ...             ...   
73385          Chicago, IL  United States             NaN           Minor   
73735  Panama City, Panama         Panama       Non-Fatal           Minor   
73761     Geneva, Eswatini       Eswatini             NaN             NaN   
75113         Patillas, PR  United States           Fatal     Substantial   
78394     Zurich, Eswatini       Eswatini       Non-Fatal             NaN   

      registration_number make  ... number_of_engines    engine_type  \
1767                N27JK  NaN  ...               1.0  Reciprocating   
1971               N8519F  NaN  ...               1.0  Reciprocating   
2458                N50RT  NaN  ...               1.0  Reciprocating   
2507               N5464E  NaN  ...               1.0  Reciprocating   
2733                N60WW  NaN  ...               1.0  Reciprocating   
...                   ...  ...  ...               ...            ...   
73385              D-AIKJ  NaN  ...               NaN            NaN   
73735                 NaN  NaN  ...               NaN            NaN   
73761                 NaN  NaN  ...               NaN            NaN   
75113               UNREG  NaN  ...               1.0  Reciprocating   
78394                 NaN  NaN  ...               NaN            NaN   

       total_fatal_injuries total_serious_injuries  total_minor_injuries  \
1767                    0.0                    1.0                   1.0   
1971                    1.0                    0.0                   0.0   
2458                    0.0                    0.0                   0.0   
2507                    0.0                    0.0                   0.0   
2733                    0.0                    2.0                   0.0   
...                     ...                    ...                   ...   
73385                   0.0                    0.0                   0.0   
73735                   0.0                    0.0                   0.0   
73761                   0.0                    0.0                   0.0   
75113                   1.0                    0.0                   0.0   
78394                   0.0                    0.0                   0.0   

       total_uninjured  weather_condition  broad_phase_of_flight  \
1767               0.0                VMC                 Cruise   
1971               0.0                VMC               Approach   
2458               1.0                VMC            Maneuvering   
2507               1.0                VMC                Landing   
2733               0.0                VMC                Takeoff   
...                ...               

In [53]:
aviation['make'].value_counts()

make
Cessna             22227
Piper              12029
CESSNA              4922
Beech               4330
PIPER               2841
                   ...  
Leonard Walters        1
Maule Air Inc.         1
Motley Vans            1
Perlick                1
ROYSE RALPH L          1
Name: count, Length: 8237, dtype: int64

Fixing the data entry issues, CESSNA and PIPER shouldn't be seperate categories

In [59]:
aviation['make'] = aviation['make'].replace('CESSNA', 'Cessna')
aviation['make'] = aviation['make'].replace('PIPER', 'Piper')

In [60]:
aviation['make'].value_counts()

make
Cessna             27149
Piper              14870
Beech               4330
Bell                2134
Boeing              1594
                   ...  
Leonard Walters        1
Maule Air Inc.         1
Motley Vans            1
Perlick                1
ROYSE RALPH L          1
Name: count, Length: 8235, dtype: int64

Filling in NaN values in the **report_status** column with most frequent report status

In [64]:
aviation['report_status'].fillna(value='Probable Cause', inplace = True)
aviation['report_status'].value_counts()
aviation['report_status'].isna().sum()

0

Filling in NaN values in **weather_condition** column with **'Missing'** value, and adding the 'UNK' and 'Unk' values to the 'Missing' category

In [66]:
aviation['weather_condition'].fillna(value='Missing', inplace = True)
aviation['weather_condition'] = aviation['weather_condition'].replace('UNK', 'Missing')
aviation['weather_condition'] = aviation['weather_condition'].replace('Unk', 'Missing')
aviation['weather_condition'].value_counts()

weather_condition
VMC        77303
IMC         5976
Missing     5610
Name: count, dtype: int64

Filling in NaN values in **broad_phase_of_flight** column with 'Missing' value, and adding 'Unknown' values to 'Missing' category

In [68]:
aviation['broad_phase_of_flight'].fillna(value='Missing', inplace = True)
aviation['broad_phase_of_flight'] = aviation['broad_phase_of_flight'].replace('Unknown', 'Missing')
aviation['broad_phase_of_flight'].value_counts()

broad_phase_of_flight
Missing        27713
Landing        15428
Takeoff        12493
Cruise         10269
Maneuvering     8144
Approach        6546
Climb           2034
Taxi            1958
Descent         1887
Go-around       1353
Standing         945
Other            119
Name: count, dtype: int64

In [72]:
aviation[aviation['injury_severity'].isna()]

event_id investigation_type accident_number  event_date  \
63918  20080111X00038           Incident      DCA08WA024  2008-01-03   
63962  20080204X00132           Accident      NYC08WA081  2008-01-16   
63987  20080304X00254           Incident      ENG08RA015  2008-01-24   
64026  20081219X65255           Incident      ENG08WA014  2008-02-03   
64131  20080409X00444           Accident      NYC08WA121  2008-02-28   
...               ...                ...             ...         ...   
90322  20221213106449           Accident      GAA22WA311  2022-12-11   
90333  20221215106462           Accident      CEN23LA064  2022-12-15   
90338  20221219106472           Accident      DCA23LA096  2022-12-18   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   

                      location        country injury_severity aircraft_damage  \
63918  Deauville Saint, France         France             NaN           Minor   
63962             Kiteni, Peru           Peru             NaN     Substantial   
63987        Kingston, Jamaica        Jamaica             NaN             NaN   
64026        Nurnberg, Germany        Germany             NaN             NaN   
64131        Lago Ranco, Chile          Chile             NaN     Substantial   
...                        ...            ...             ...             ...   
90322                Kildare,         Ireland             NaN       Destroyed   
90333            Patterson, LA  United States             NaN             NaN   
90338              Kahului, HI  United States             NaN             NaN   
90344              Hampton, NH  United States             NaN             NaN   
90346               Morgan, UT  United States             NaN             NaN   

      registration_number                         make  ... number_of_engines  \
63918              CN-RMX                       Boeing  ...               NaN   
63962             OB-1850                         Bell  ...               NaN   
63987              6Y-JRG                       Cessna  ...               NaN   
64026              LN-RRR                       BOEING  ...               2.0   
64131              CC-PIL                        Piper  ...               NaN   
...                   ...                          ...  ...               ...   
90322              N999RL  ROBINSON HELICOPTER COMPANY  ...               1.0   
90333              N414RL                         BELL  ...               NaN   
90338              N393HA                       AIRBUS  ...               NaN   
90344              N2895Z                     BELLANCA  ...               NaN   
90346              N210CU                       Cessna  ...               NaN   

      engine_type  total_fatal_injuries total_serious_injuries  \
63918         NaN                   0.0                    0.0   
63962         NaN                   0.0                    0.0   
63987         NaN                   0.0                    0.0   
64026         NaN                   0.0                    0.0   
64131         NaN                   0.0                    0.0   
...           ...                   ...                    ...   
90322         NaN                   0.0                    0.0   
90333         NaN                   0.0                    0.0   
90338         NaN                   0.0                    0.0   
90344         NaN                   0.0                    0.0   
90346         NaN                   0.0                    0.0   

       total_minor_injuries  total_uninjured  weather_condition  \
63918                   0.0              0.0            Missing   
63962                   0.0              0.0            Missing   
63987                   0.0              0.0            Missing   
64026                   0.0              0.0            Missing   
64131                   0.0              0.0                VMC   
...    

Looks like the rows with NaN values for injury severity have no injury data, so I'm going to drop those rows

In [77]:
aviation.dropna(subset=['injury_severity'], inplace = True)
aviation['injury_severity'].isna().sum()

0

Filling in the NaN values with "Missing" value and adding "Unknown" values to that category for the rest of the categorical columns

In [80]:
aviation['aircraft_damage'].value_counts()

aircraft_damage
Substantial    63964
Destroyed      18561
Minor           2583
Unknown           92
Name: count, dtype: int64

In [82]:
aviation['aircraft_damage'].fillna(value="Missing", inplace = True)
aviation['aircraft_damage'] = aviation['aircraft_damage'].replace('Unknown', 'Missing')
aviation['aircraft_damage'].value_counts()


aircraft_damage
Substantial    63964
Destroyed      18561
Missing         2781
Minor           2583
Name: count, dtype: int64

In [83]:
aviation['engine_type'].value_counts()

engine_type
Reciprocating      69503
Turbo Shaft         3602
Turbo Prop          3372
Turbo Fan           2360
Unknown             2049
Turbo Jet            698
Electric               8
LR                     2
NONE                   2
Hybrid Rocket          1
Geared Turbofan        1
UNK                    1
Name: count, dtype: int64

In [84]:
aviation['engine_type'].fillna(value="Missing", inplace = True)
aviation['engine_type'] = aviation['engine_type'].replace('Unknown', 'Missing')
aviation['engine_type'] = aviation['engine_type'].replace('UNK', 'Missing')
aviation['engine_type'] = aviation['engine_type'].replace('NONE', 'Missing')
aviation['engine_type'].value_counts()


engine_type
Reciprocating      69503
Missing             8342
Turbo Shaft         3602
Turbo Prop          3372
Turbo Fan           2360
Turbo Jet            698
Electric               8
LR                     2
Hybrid Rocket          1
Geared Turbofan        1
Name: count, dtype: int64

In [85]:
aviation.isna().sum()

event_id                      0
investigation_type            0
accident_number               0
event_date                    0
location                     52
country                     226
injury_severity               0
aircraft_damage               0
registration_number        1359
make                         46
model                        72
amateur_built               102
number_of_engines          5363
engine_type                   0
total_fatal_injuries      11401
total_serious_injuries    12510
total_minor_injuries      11933
total_uninjured            5912
weather_condition             0
broad_phase_of_flight         0
report_status                 0
publication_date          14952
dtype: int64

In [90]:
aviation['total_fatal_injuries'].value_counts()

total_fatal_injuries
0.0      58675
1.0       8883
2.0       5173
3.0       1589
4.0       1103
         ...  
156.0        1
68.0         1
31.0         1
115.0        1
176.0        1
Name: count, Length: 125, dtype: int64